### ISSUE TITLE
fatal error during parsing

### ISSUE THREAD

**Issue by smacker** [bblfsh/scala-client#77]
The easiest way to reproduce bug is to replace current java file with my go file (see links below) in `BblfshClientParseTest`. And then run `./sbt test`. It produces an error:

```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x000000010f71c270, pid=30822, tid=0x000000000000420f
#
# JRE version: Java(TM) SE Runtime Environment (8.0_152-b16) (build 1.8.0_152-b16)
# Java VM: Java HotSpot(TM) 64-Bit Server VM (25.152-b16 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# V  [libjvm.dylib+0x31c270]  jni_GetObjectField+0x6e
#
# Core dump written. Default location: /cores/core or core.30822
#
# An error report file with more information is saved as:
# /Users/smacker/Work/client-scala/hs_err_pid30822.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
```

Go file: https://gist.github.com/smacker/60e16f66aa4198eee533b229b11986a4
Log: https://gist.github.com/smacker/817eeb635e2ff896657cebb2aac5b3a9

I also tried to recompile client with libuast v1.9.5 but it didn't help.

**Comment by smacker** [comment]
Oh. I also see errors in bblfshd (`bblfsh/bblfshd:v2.9.2-drivers`):
```
[2019-01-21T20:18:09Z] ERROR request processed content 2260 bytes, status Fatal elapsed=2.9835ms filename=/Users/smacker/Work/gemini/src/test/resources/consumer.go language=go
```

**Comment by dennwc** [comment]
I'm not familiar with this codebase, but this kind of response from `bblfshd` might be related to either driver failure (which is unlikely) or the protocol error (but new versions still serve v1).

Can you please try to parse this file directly with this `bblfshd` version to see if it fails?

**Comment by smacker** [comment]
```
root@fdf84e7b0faf:/# bblfshctl parse /consumer.go
Status: Ok
Elapsed: 64.6677ms
...
```

Could you try to use the same file in the test on your machine? I'm not sure if it's related to macOs anymore.

**Comment by bzz** [comment]
So, JVM is shutdown by `SIGSEGV` and if the file is successfully parsed by bblfsh. 

From that + logs of top of the stack `C  [libscalauast_86840640813667747.dylib+0x1e6e]  ReadStr(_jobject* const*, char const*)+0x11e` I would take wild guess and say it's a [cross-compiled native libuast](https://github.com/bblfsh/client-scala/blob/master/build.sbt#L181) that is segfaulting on a new version of macOS, same as in https://github.com/src-d/enry/pull/190

Let me look today into what is the best way to address those consistently.

**Comment by smacker** [comment]
Not sure why you guess it as a cross-compilation issue.
Running tests requires local building of libuast. So the error from above happens with libuast compiled locally. Full list of commands:
```
$ git clean -ffdx

$ git diff
--- a/src/test/scala/org/bblf/client/BblfshClientParseTest.scala
+++ b/src/test/scala/org/bblf/client/BblfshClientParseTest.scala
-  val fileName = "src/test/resources/SampleJavaFile.java"
+  //val fileName = "src/test/resources/SampleJavaFile.java"
+  val fileName = "/Users/smacker/Work/gemini/src/test/resources/consumer.go"

$ ./sbt -java-home /Library/Java/JavaVirtualMachines/jdk1.8.0_152.jdk/Contents/Home assembly test
<segfault>
```

**Comment by bzz** [comment]
Why is that necessary to provide `-java-home` ? Do you have multiple ones in your configuration?

> Not sure why you guess it as a cross-compilation issue.

As explain above, I was just sharing an idea, as that kind of collaboration helps in faster debugging and so far I have not seen any other constructive hypothesis posted. 

Please do not hesitate to try doing that as well 😉

So far I do not have machine with Mojave to reproduce this, so will it open with a low priority.

**Comment by smacker** [comment]
> Why is that necessary to provide -java-home ?

because the script in `build.sbt` can't find java headers. If you check the script it looks into env variable `JAVA_HOME` which is empty by default on macOs (at least in all my installations) and assumes linux default path `/usr/lib/jvm/java-8-openjdk-amd64`.

> Do you have multiple ones in your configuration? Could that be that different versions of libjvm are used at compilation/runtime?

I have only one JVM.

> As explain above, I was just sharing an idea, as that kind of collaboration helps in faster debugging and so far I have not seen any other constructive hypothesis posted.

As I mentioned above it might be non-mojave problem. It's just harder to test for me than for you. Though I did it using docker:

```
$ docker run -it --link b2059328015c --rm openjdk:8 bash
root@23706899d199:/# git clone https://github.com/bblfsh/client-scala.git
root@23706899d199:/# cd client-scala/
root@23706899d199:/client-scala# apt-get update
root@23706899d199:/client-scala# apt-get install libxml2-dev g++ gcc
root@23706899d199:/client-scala# ./sbt assembly
<update test to use b2059328015c>
root@23706899d199:/client-scala# ./sbt test
[info] All tests passed

<upload file inside docker and update test to use it>
root@5d094769b9dc:/client-scala# ./sbt test
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fd83fc38c4d, pid=1714, tid=0x00007fd7f35fd700
#
# JRE version: OpenJDK Runtime Environment (8.0_181-b13) (build 1.8.0_181-8u181-b13-2~deb9u1-b13)
# Java VM: OpenJDK 64-Bit Server VM (25.181-b13 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# V  [libjvm.so+0x67ec4d]
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /client-scala/hs_err_pid1714.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
Aborted
```

> keep it open but with a low priority

it's fine

**Comment by ncordon** [comment]
This does not appear to be an issue anymore. Code tested:

```scala
import scala.io.Source
import org.bblfsh.client.v2.BblfshClient, BblfshClient._
import gopkg.in.bblfsh.sdk.v2.protocol.driver.Mode

val client = BblfshClient("localhost", 9432)
val filename = "/run/media/nuwanda/Datos/sourced/gemini/src/test/resources/consumer.go"
val resp = client.parse(filename, fileContent, Mode.SEMANTIC)
val ctx = resp.get()

val resp = client.parse(filename, fileContent, Mode.ANNOTATED)
val ctx = resp.get()

val resp = client.parse(filename, fileContent, Mode.NATIVE)
val ctx = resp.get()
```

Please, feel free to reopen if it has not been entirely fixed.
